examples/fargate/main.tf

```diff
@@ -61,7 +61,8 @@ module "eks" {
   source          = "../.."
   cluster_name    = local.cluster_name
   cluster_version = "1.20"
-  subnets         = module.vpc.private_subnets
+  subnets         = concat(module.vpc.private_subnets,module.vpc.public_subnets)
```

Comment:
can you fix a little this example as this is quite bad practice to put things in public subnet on EKS

Reply:
Why would it be bad practice? It would be a requirement to allow a public facing lb. The fargate profiles are still private, as they should be, but allowing for external access to some eks services is what I see as the use case for this change.

Reply:
in general best practice is putting all workers inside private subnets, publicly faced loadbalancers will be automatically created in public subnets and set backend members into private subnets. 

Something like here:
![image](https://user-images.githubusercontent.com/20227477/132222387-473d3062-6c16-4860-a75d-8d31ea6aade3.png)

To place a loadbalancer into public subnet you don`t require have there EKS Control plane endpoint.

Reply:
So in my case all of my workers would be in private nodes, thats the fargate_profiles. Fargate would fail if I tried to put it in a public subnet. This allows for private workers, but for the eks cluster to have public subnets attached, which the aws ingress controller requires, as when asked to provide external access to a service running on fargate it will look for a public subnet for the load balancer that it will create. 

In that diagram, where would the cluster be? If that would only be the fargate section, then it would require LBs to be made seperately from deployments, and then linked together rather than being able to use an ingress controller to handle creation of ingress depending on annotations. If the cluster in the diagram would include the load balancer then I don't see a way for the ingress controller to create it in a public subnet without that subnet being attached to the cluster.

Reply:
Best practice for cluster and workers it to put them into private subnets. 
You can try to deploy https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/examples/basic/main.tf  and then push to kubernetes for example https://artifacthub.io/packages/helm/bitnami/nginx which will create Loadbalancer service type. Then you will see that Loadbalancer will be publicly available but your workers will be still in private subnets under the NAT